### PR TITLE
PRIME-1563 Angular Upgrade v12

### PR DESCRIPTION
**Pull request by mtpultz**
Angular has been updated from v10 through to v12.  This includes Angular Material, as well as removal of TSLint and migration to ESLint.  Hopefully this reduces or gets rid of our memory issues in VSCode around TSLint scanning all the project files.

You can also update your local install of the Angular CLI by typing:

```
npm uninstall -g @angular/cli
npm install -g @angular/cli@latest
```

If you run into problems locally after running `npm install` try `npm cache clear` or simply delete the `node_modules` folder before running `npm install` again.

### CODE REVIEW

File: prime-angular-frontend/openshift.dockerfile [hunk 1]
```diff
@@ -34,10 +34,11 @@ COPY package.json package-lock.json ./
 COPY . .
 
 # Install Angular CLI
+RUN npm set unsafe-perm true
 RUN npm install -g @angular/cli
 # Install dependencies
 RUN npm ci
-# Add application
+# Add applicationz
```
Comment (AMiKaH) [inline prime-angular-frontend/openshift.dockerfile:41]: I know this is just a comment, but there's an extra 'z' not sure if typo or supposed to be like that :)

Comment (mtpultz) [reply prime-angular-frontend/openshift.dockerfile:41]: Total typo! Thanks :)

**Review by johnnyl6 — APPROVED** [review]
ESLint, yay!👏

**Review by AMiKaH — APPROVED** [review]

**Review by neophyte57 — APPROVED** [review]
I used PR #1518 against this PR environment and all tests could past.  The e2e tests are limited but still provide significant coverage.

**Review by Paahn — APPROVED** [review]
this is the way

**Pull request merged**